core/tests/protractor/editorAndPlayer.js

```diff
@@ -26,9 +26,69 @@ var ExplorationPlayerPage =
 
 describe('Full exploration editor', function() {
   var explorationPlayerPage = null;
+  var parentId2 = null;
+  var refresherExplorationId = null;
+  var currentExplorationId = null;
+  var pathname = null;
+  var parentId1 = null;
 
   beforeAll(function() {
     explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+    users.createUser('user4@editorAndPlayer.com', 'user4EditorAndPlayer');
```

Comment:
Why put all this stuff in beforeAll? It doesn't need to run for all the tests.

Rather, this should be a separate test in its own right, similar to the others below.

Reply:
I did this because, setting parentId1 and parentId2 asynchronously using browser.getCurrentUrl().then() didn't work when they were in same test, as the variables just became null and values were set after their expect was called.
I thought beforeAll() would be called just once at the beginning of the test, will try something else then.

Reply:
Ah, OK, thanks for explaining. Yep, beforeAll() gets run before every single test, so some other solution is needed.

Reply:
Can I have a chain of general.getExplorationIdFromEditor() within each promise?

Reply:
I'm not sure what you mean; could you show some sample code?

Reply:
general.getExplorationIdFromEditor().then(function(explorationId) {
        parentId1 = explorationId;
        workflow.createAndPublishExploration( 'Title 2', 'Algebra', 'It is a sample exploration.');
        general.getExplorationIdFromEditor().then(function(explorationId) {
          parentId2 = explorationId;
             and so on.
        });
    });

Reply:
Good question. It's a little icky, but I think it's probably OK if there's no synchronous alternative.

Reply:
Ok, I'll look, but I don't think there are any synchronous ways to do this.
I could have a dummy test that just sets the ids without checking anything and the actual test that does the redirection and checks if they are correct.

Reply:
What do you mean by "sets the ids without checking anything"? I don't understand.

Reply:
I meant instead of what I did with putting those in beforeAll(), I would put it in another test before this one, so that all variables will get the values (only thing is that 'test' won't actually test anything, just set id values to all the variables).

Reply:
Oh, I see. No, don't do that. The tests should be independent. What you suggested before that (the nested promises) is OK.

Reply:
Ok, Done